soap/client.go

```diff
@@ -185,6 +189,17 @@ func (c *Client) RoundTripSoap12(action string, in, out Message) error {
 	return doRoundTrip(c, headerFunc, in, out)
 }
 
+//Detailed soap http error
+type HttpError struct {
```

Comment:
Please ensure it conforms to golint by having comment begins with the name of the element it describes.  Thanks!